### ISSUE TITLE
[Bug]: Terraform does not import DataSync S3 location with a required argument

### ISSUE THREAD

**Issue by markoeremija** [hashicorp/terraform-provider-aws#31678]
### Terraform Core Version

1.4.2

### AWS Provider Version

4.67.0

### Affected Resource(s)

aws_datasync_location_s3

### Expected Behavior

When importing an existing AWS DataSync S3 location to the terraform state, it would be expected to import also the argument marked as required in the documentation, i.e., [`s3_bucket_arn`](https://registry.terraform.io/providers/hashicorp/aws/4.67.0/docs/resources/datasync_location_s3#s3_bucket_arn).

### Actual Behavior

The `terrform import` command completes successfully (based on the example from the [documentation](https://registry.terraform.io/providers/hashicorp/aws/4.67.0/docs/resources/datasync_location_s3#import)):

```
> terraform import module.datasync.aws_datasync_location_s3.source arn:aws:datasync:<region>:<account-id>:location/<loc-id> 

Acquiring state lock. This may take a few moments...
module.datasync.aws_datasync_location_s3.source: Importing from ID "arn:aws:datasync:<region>:<account-id>:location/<loc-id>"...
module.datasync.aws_datasync_location_s3.source: Import prepared!
  Prepared aws_datasync_location_s3 for import
module.datasync.aws_datasync_location_s3.source: Refreshing state... [id=arn:aws:datasync:<region>:<account-id>:location/<loc-id>]

Import successful!

The resources that were imported are shown above. These resources are now in
your Terraform state and will henceforth be managed by Terraform.
```
The subsequent `terraform plan` shows that terraform wants to recreate everything (including the DataSync task using this source, omitted for brevity) because the `s3_bucket_arn` argument was added:

```
  # module.datasync.aws_datasync_location_s3.source must be replaced
-/+ resource "aws_datasync_location_s3" "source" {
      - agent_arns       = [] -> null
      ~ arn              = "arn:aws:datasync:<region>:<account>:location/<loc-id>" -> (known after apply)
      ~ id               = "arn:aws:datasync:<region>:<account>:location/<loc-id>" -> (known after apply)
      + s3_bucket_arn    = "arn:aws:s3:::<bucket-name>" # forces replacement
      ~ s3_storage_class = "STANDARD" -> (known after apply)
      ~ uri              = "s3://<bucket-name>/path/to/copy/from/" -> (known after apply)
        # (2 unchanged attributes hidden)

        # (1 unchanged block hidden)
    }
```

The resource definition in the module is as follows:

```hcl
resource "aws_datasync_location_s3" "source" {
  s3_bucket_arn = var.source_s3_bucket_arn
  subdirectory  = var.source_subdirectory

  s3_config {
    bucket_access_role_arn = var.bucket_access_role_arn
  }
}
```

Using `terraform state show module.datasync.aws_datasync_location_s3.source` gives the following output:

```
# module.datasync.aws_datasync_location_s3.source:
resource "aws_datasync_location_s3" "source" {
    agent_arns       = []
    arn              = "arn:aws:datasync:<region>:<account>:location/<loc-id>"
    id               = "arn:aws:datasync:<region>:<account>:location/<loc-id>"
    s3_storage_class = "STANDARD"
    subdirectory     = "/path/to/copy/from/"
    uri              = "s3://<bucket-name>/path/to/copy/from/"

    s3_config {
        bucket_access_role_arn = "arn:aws:iam::<account>:role/<datasync-role-name>"
    }
}
```

As can be seen from the `terraform state show` output, there is no `s3_bucket_arn` argument. Currently, the workaround is to use `lifecycle` block to ignore changes to the `s3_bucket_arn` argument:

```hcl
resource "aws_datasync_location_s3" "source" {
  s3_bucket_arn = var.source_s3_bucket_arn
  subdirectory  = var.source_subdirectory

  s3_config {
    bucket_access_role_arn = var.bucket_access_role_arn
  }

  lifecycle {
    ignore_changes = [
      s3_bucket_arn
    ]
  }
}
```

This approach solves the issue with terraform wanting to recreate everything.

### Terraform Configuration Files

```hcl
resource "aws_datasync_location_s3" "source" {
  s3_bucket_arn = var.source_s3_bucket_arn
  subdirectory  = var.source_subdirectory

  s3_config {
    bucket_access_role_arn = var.bucket_access_role_arn
  }

  lifecycle {
    ignore_changes = [
      s3_bucket_arn
    ]
  }
}

resource "aws_datasync_location_s3" "destination" {
  s3_bucket_arn = var.destination_s3_bucket_arn
  subdirectory  = var.destination_subdirectory

  s3_config {
    bucket_access_role_arn = var.bucket_access_role_arn
  }

  lifecycle {
    ignore_changes = [
      s3_bucket_arn
    ]
  }
}

resource "aws_datasync_task" "this" {
  name                     = var.datasync_task_name
  source_location_arn      = aws_datasync_location_s3.source.arn
  destination_location_arn = aws_datasync_location_s3.destination.arn
  cloudwatch_log_group_arn = var.cloudwatch_log_group_arn

  dynamic "options" {
    for_each = var.use_options ? [1] : []
    content {
      gid               = lookup(var.options, "gid", null)
      log_level         = lookup(var.options, "log_level", null)
      posix_permissions = lookup(var.options, "posix_permissions", null)
      uid               = lookup(var.options, "uid", null)
    }
  }

  dynamic "schedule" {
    for_each = var.use_schedule ? [1] : []
    content {
      schedule_expression = var.schedule_expression
    }
  }
}
```

### Steps to Reproduce

- Create AWS DataSync source and destination S3 locations along with a task in AWS console.
- Import those resources into terraform state, following the official documentation for each of the resources.
- Run `terraform plan`.
- Verify that the resources will be recreated due to the fact that `s3_bucket_arn` is missing after the import.

### Would you like to implement a fix?

No

**Comment by acwwat** [comment]
@markoeremija This should be fixed by #36072 which went into v5.42.0 of the provider. Please give it a try and close this ticket once verified. Thanks.

**Comment by markoeremija** [comment]
@acwwat Thanks for letting me know it's fixed, I no longer have access to the codebase where I found the bug, but I'll close it anyway.
